client/test/Fred.go

```diff
@@ -0,0 +1,24 @@
+package test
```

Comment:
I think the test can go in one file. Also note that go files start with a lowercase letter.